pkg/persistence/account/validate.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package account
+
+import "fmt"
+
+// Validate checks the references in the provided AMResources instance to ensure
+// that all referenced groups and policies exist. It iterates through the users,
+// environment policies, and account policies, validating their references.
+func Validate(res *AMResources) error {
+	for _, user := range res.Users {
+		for _, groupRef := range user.Groups {
+			if err := refCheck(groupRef, res.GroupExists); err != nil {
```

Comment:
So for each user with group references, we iterate each group and check if the referenced ID is found.
And then we iterate the groups once again to iterate over all policies twice for each of them (once for env policies and once for account).

Couldn't we achieve the same with fewer iterations if we reversed the order of the checks?

1. Iterate res.Policies once, build set of IDs
2. Iterate res.Groups once, build set of IDs & check for each env and account policy if it's known/in the policy ID set
3. Iterate res.Users once, check for each group reference if it's in the group ID set

Reply:
Or alternatively if it makes sense store the objects in a id->object maps inside AMResources already, instead of slices.

With this we could already produce early errors for duplicate IDs, which I've just noticed is missing from validation at the moment?

Reply:
i didn't care about performance. But sure we can do that :)

Reply:
😱 You what?! I'm shocked @warber... shocked... 

Seriously though, I think it's really worth closely thinking about performance for this. 
If a user wants to configure account permissions as code, they're likely doing it at a larger scale. For a few users and default permissions, they probably won't start with config-as-code

Reply:
makes sense, didn't expect account resources on a larger scale, will update the pr